### ISSUE TITLE
Support partial field mapping transformation for nested fields

### ISSUE THREAD

**Issue by Samrose-Ahmed** [SigmaHQ/pySigma#61]
Field mapping transformations currently map directly on the key. If the key isn't found, a mapping isn't applied. But sometimes you have dynamic keys that don't make sense to map out explicitly. 

E.g., I was writing a pipeline for ECS CloudTrail and the `responseElements` field is a very polymorphic field that can't really be mapped out fully. It would be good to map, [for example](https://github.com/SigmaHQ/sigma/blob/master/rules/cloud/aws/aws_rds_public_db_restore.yml#L16), `responseElements.publiclyAccessible` to  `aws.cloudtrail.response_elements.publiclyAccessible`, based on [the mapping](https://github.com/SigmaHQ/sigma/blob/master/tools/config/ecs-cloudtrail.yml#L31) `"responseElements": "aws.cloudtrail.response_elements"`, if a direct mapping isn't found.

I can use a new Transformation class for this if you don't think this should be supported by default. The change should be fairly simple [here](https://github.com/SigmaHQ/pySigma/blob/main/sigma/processing/transformations.py#L191).

**Comment by thomaspatzke** [comment]
Sounds useful! What do you think about a regular expression field naming transformation? For your example something like this:

```
RegexFieldMappingTransformation("responseElements\.(\\w+)", "aws.cloudtrail.response_elements.\\1")
```

**Comment by frack113** [comment]
I think more to a generic name (even if it the same logic/code )
NestedFieldMappingTransformation("nested_ori", "nested_dest")
   nested_dest.\\1 <== (nested_ori.(\\w+)

**Comment by thomaspatzke** [comment]
I don't see the "nested" as generic name because it is one of many use cases that can be done with this. A transformation `foo\.some(thing).bar` to `\1.bla` would be possible, too. Implementation would be done with regular expressions anyways and I find it preferable to call the features by their name 😉

**Comment by Samrose-Ahmed** [comment]
I think regex is cool to support for some usecases, but I think this should be supported by default in FieldMappingTransformation.  We're mapping some nested field in the source hirearchy to a field in the output hirearchy, we shouldn't just support leaf field names.  Like if you have a mapping for `a.b.c -> x.y.z`, why not map  `a.b.c.d` to `x.y.z.d`? Let me know what you think.

**Comment by thomaspatzke** [comment]
I prefer to keep the semantics of FieldMappingTransformation simple because I fear that overloading a simple transformation with functionality that is not wanted in most cases could cause unwanted and hard to debug side-effects. What if a mapping rule `a -> b` unintentionally causes that a mapping `a.c.d -> b.c.d` is applied? So there should be a flag to disable/enable this behavior for cases where the users explicitely don't or do want it. And there we are, I believe it is easier and cleaner to keep this as separate transformations.

Instead (or in addition to, I've planned to introduce it anyways) the proposed regular expression field name transformation I would just add a FieldPrefixMappingTransformation, that does what you describe, e.g. `FieldPrefixMappingTransformation("a.b.c.", "x.y.z.")`.

By the way...for Sigma field names are simple strings and no assumptions regarding hierarchical naming schemes etc. were made to keep it as generic as possible. If we would made such assumptions, e.g. about the dot as hierarchical separator, this would just result in additional complexity. So I would implement this as simple string replacement transformation.

**Comment by Samrose-Ahmed** [comment]
That's fair enough, either the regex or the prefix is fine.

**Comment by thomaspatzke** [comment]
Implemented it, will be contained in pySigma 0.8.9 and sigma-cli 0.5.3.
